src/ramses_rf/device/hvac.py

```diff
@@ -383,6 +383,19 @@ def indoor_humidity(self) -> float | None:
 
     @property
     def indoor_temp(self) -> float | None:
+        if Code._12A0 in self._msgs:
+            if isinstance(
+                self._msgs[Code._12A0].payload, list
+            ):  # FAN Ventura sends RH/temps as a list, use element [0] for indoor_temp
+                for k, v in self._msgs[Code._12A0].payload[0].items():
+                    if k == SZ_TEMPERATURE:
+                        return float(v)
+            for k, v in self._msgs[Code._12A0].payload.items():
+                if (
+                    k == SZ_TEMPERATURE
+                ):  # ClimaRad minibox FAN sends (indoor) temp in 12A0
+                    return float(v)
+
```

Comment:
This doesn't look right - at least use a guard clause:
```python
if Code._12A0 not in self._msgs:
    return None
```

Reply:
This part of the code is a mess... Have a look at switching over to `MessageIndex` (see **ramses_rf/database.py**).

Reply:
added 2 times